src/OrchardCore/OrchardCore.Recipes.Abstractions/Services/IRecipeExecutor.cs

```diff
@@ -1,10 +1,11 @@
-﻿using System.Threading.Tasks;
+using System.Threading.Tasks;
 using OrchardCore.Recipes.Models;
 
 namespace OrchardCore.Recipes.Services
 {
     public interface IRecipeExecutor
     {
         Task<string> ExecuteAsync(string executionId, RecipeDescriptor recipeDescriptor, object environment);
+        Task<string> ExecuteMigrationAsync(string executionId, RecipeDescriptor recipeDescriptor, object environment);
```

Comment:
OK ... it will need a boolean to indicate whether this is a migration recipe or not ... if you are OK with that. I know I am asking you to make a very granular decision here, but here goes. 3 choices so pick one:
1. Method param and either:
   a. specify default value in method signature
   b. change all invocations to specify param value
2. IsMigrationRecipe prop added to RecipeDescritor (and maybe add check to harvester code? - that way all recipe files could be placed in Recipes folder)
3. IsMigration prop added to environment object - not sure if that's the right place for it